specifier.go

```diff
@@ -72,11 +72,62 @@ func NewSpecifiersWithSanitizer(v string, sanitizer func(string) string, opts ..
 	return newSpecifiers(v, sanitizer, opts...)
 }
 
+// NewPermisiveSpecifiers parses a given specifier and returns a new instance of Specifiers. It is
+// similar to NewSpecifiersWithSanitizer but allows dash '-' characters in a version and replaces them
+// with periods.
+func NewPermisiveSpecifiers(v string, sanitizer func(string) string, opts ...SpecifierOption) (Specifiers, error) {
```

Comment:
I'll be ok with a new separate function just for R, but we should document or rename this so it's very clear that this is a special case only meant for R specifiers.